nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-jetty/src/main/java/org/apache/nifi/web/server/JettyServer.java

```diff
@@ -607,6 +612,25 @@ private WebAppContext loadWar(final File warFile, final String contextPath, fina
         return webappContext;
     }
 
+    private void addContentLengthFilter(String contextPath, String pathSpec, WebAppContext webappContext) {
+        final FilterHolder holder = new FilterHolder(ContentLengthFilter.class);
+        final Map<String, String> largePaths = props.getWebMaxContentSizeLargePaths();
+        int size;
+
+        if (!largePaths.containsValue(contextPath)) {
+            size = DataUnit.parseDataSize(props.getWebMaxContentSize(), DataUnit.B).intValue();
```

Comment:
What happens if the max content size is not parseable as a data unit? In the unit tests, these properties accepted "size value" and "large size value" without complaint, so I don't believe there's any internal validation occurring.

Reply:
I've updated the method that uses the values to handle parsing exceptions and values out of range.